torch/distributed/distributed_c10d.py

```diff
@@ -1465,6 +1466,44 @@ def reduce_scatter(output,
         work.wait()
 
 
+def alltoall(output,
+                   input,
+                   group=group.WORLD,
+                   async_op=False):
+    """
+    Each process splits input tensor equally and then scatters the split list to all processes in a group.
+    Then concatenate the received tensors from all the processes in the group and return single output tensor.
+
```

Comment:
i think it might be helpful to provide a picture with 16 tensors and 4 workers here as well

Reply:
Updated commit with alltoall example.